libnd4j/include/ops/declarable/generic/helpers/impl/convolutions.cpp

```diff
@@ -737,56 +768,62 @@ void ConvolutionUtils<T>::maxPool3dFrame(NDArray<T>& input, NDArray<T>& output,
     const int oD = output.sizeAt(2);
     const int oH = output.sizeAt(3);
     const int oW = output.sizeAt(4);    
-
-    int k;
-
-#pragma omp parallel for private(k)
-    for (k = 0; k < iC; k++) {
     
-        /* loop over output */
-        int i, j, ti;
-        for (ti = 0; ti < oD; ti++) {
-            for (i = 0; i < oH; i++) {
-                for (j = 0; j < oW; j++){
+    const int iStride2 = iH * iW;
+    const int oStride2 = oH * oW;
+    const int iStride1 = iD * iStride2;
+    const int oStride1 = oD * oStride2;
+    const int iStride0 = iC * iStride1;
```

Comment:
use auto or Nd4jLong here please. Convolution stuff was the reason behind int -> long migration, exactly due to possibility of getting strides > MAX_INT.

Reply:
done